x-pack/plugins/apm/common/utils/formatters/alert_url.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { stringify } from 'querystring';
+import { ENVIRONMENT_ALL } from '../../environment_filter_values';
+
+const format = ({
+  pathname,
+  query,
+}: {
+  pathname: string;
+  query: Record<string, any>;
+}): string => {
+  return `${pathname}?${stringify(query)}`;
+};
+
+export const getAlertUrlErrorCount = (serviceName: any[], serviceEnv: any[]) =>
+  format({
+    pathname: `/app/apm/services/${String(serviceName[0])}/errors`,
+    query: {
+      ...(serviceEnv?.[0]
+        ? { environment: String(serviceEnv[0]) }
+        : { environment: ENVIRONMENT_ALL.value }),
+    },
+  });
+
+export const getAlertUrlTransactionDuration = (
```

Comment:
@gbamparop, you are totally right! And I thought even going further and using one function for all e.g., we could also include the Error Count function, it doesn't have `transaction` and its route ends with `/error`, but it's easy to add some conditions and have one for all.

 However, I didn't go that path because:
1- I wanted only to use the existing functions in the backend, so I tried to stay in the scope of the PR, not starting a bigger refactoring, especially this PR planned to be shipped in v8.2 
2- As you mentioned, they are very similar,  but they were separate functions in the first place, so I think they have been kept like this for a reason.

If you like, we can create a separate Issue to refactor them and see if it's ok to have one function.

Reply:
Thank you for looking into this! I agree that the scope of this PR shouldn't be extended to cover more tech debt but it seems that before they were just defined in place, so now that helper functions are created I believe it makes sense to extract only what is necessary. 

I've checked the functions again and two are the same (apart from the Non-null assertion) and the other one doesn't convert the params to a string before passing it to `format`.

Reply:
Hi, I would like to add that this PR is also blocking the merge of https://github.com/elastic/kibana/pull/128249, also planned to be shipped in v8.2, FF is next Tuesday, we still have time, but maybe creating a separate issue for the refactoring is a good idea.

Reply:
Sure, we could do if you think it makes sense

Reply:
I just refactored the functions as @gbamparop requested.
cc @MiriamAparicio

Reply:
Great, looks more concise now thanks!